frontend/src/__tests__/api.test.js

```diff
@@ -269,11 +435,81 @@ function positionsTests(api = { apiGET, apiPOST }) {
-    it.todo("update an instructor");
-    it.todo("add and remove instructor from position");
+    let session = null,
+        position = null,
+        instructor = null;
+    // set up a session to be available before tests run
+    beforeAll(async () => {
+        // this session will be available for all tests
+        session = await addSession({ apiGET, apiPOST });
+        position = await addPosition({ apiGET, apiPOST }, session);
+    });
+    // delete the session after the tests run
+    afterAll(async () => {
+        await deletePosition({ apiGET, apiPOST }, position);
+        await deleteSession({ apiGET, apiPOST }, session);
+    });
+
+    it("get instructors", async () => {
+        const resp = await apiGET("/instructors");
+        expect(resp).toMatchObject({ status: "success" });
+        checkPropTypes(PropTypes.arrayOf(instructorPropTypes), resp.payload);
+    });
+
+    it("get instructors for session", async () => {
+        const resp = await apiGET(`/sessions/${session.id}/instructors`);
+        expect(resp).toMatchObject({ status: "success" });
+        checkPropTypes(PropTypes.arrayOf(instructorPropTypes), resp.payload);
+    });
+
+    it("create instructor", async () => {
+        const newInstructorData = {
+            first_name: "Anand",
+            last_name: "Liu",
+            email: "anand.liu.sample@utoronto.ca",
+            utorid: "anandl"
+        };
+
+        // create a new instructor
+        const resp1 = await apiPOST(`/instructors`, newInstructorData);
+        expect(resp1).toMatchObject({ status: "success" });
+        expect(resp1.payload).toMatchObject(newInstructorData);
+
+        // make sure instructor is on instructor list
+        const resp2 = await apiGET(`/instructors`);
+        expect(resp2).toMatchObject({ status: "success" });
+        expect(resp2.payload).toContainObject(newInstructorData);
+
+        // set instructor to used by later test
+        instructor = resp1.payload;
+    });
+
+    it("add instrutor to position", async () => {
+        const resp = await apiPOST(`/positions/${position.id}/add_instructor`, {
+            id: instructor.id,
+            position_id: position.id
+        });
+        expect(resp).toMatchObject({ status: "success" });
+        expect(resp.payload).toMatchObject(instructor);
+    });
+
+    it("update an instructor", async () => {
+        const updateInstructorData = {
+            id: instructor.id,
+            email: "anand.liu@gmail.com"
+        };
+
+        // update the instructor
+        const resp = await apiPOST(`/instructors`, updateInstructorData);
+        expect(resp).toMatchObject({ status: "success" });
+        expect(resp.payload).toMatchObject(updateInstructorData);
+    });
+
+    // delete an instructor
+    it("delete instructor", async () => {
+        const resp = await apiPOST(`/instructors/delete`, instructor);
+        expect(resp).toMatchObject({ status: "success" });
```

Comment:
Get the instructor list and make sure they are not there. We also need to delete an instructor from a session.